### PR TITLE
Uniques040

### PR DESCRIPTION
**Pull request by MysticJay**
This PR extends UNIQUES to allow keeping track of portals that have been scanned for "Scout Controller" (scoutControlled) or visited by a "Drone" (dronedVisited).

In general the user has to mark scoutControlled or droneVisited portals manually. Additionally UNIQUES will analyze the COMM on ALL and ALERTS (requires a core change to CHAT.JS) for messages that indicate a previous actions withthe portal. The information will be added immediately to the portal-list to those portals visible on the map. For unresolved portals the information (LatLonE6) will be stored in a separate list called missedPortals. As soon as one of the missed portals appears on the map the information will be added to portal-list as well.

To improve the hitrate, when resolving the information the plugin Cache-Portals-On-Map and Offle can assist to resolve LatLonE6 to guid.

UNIQUES now has a dialog that allows to jump to those locations listed in missedLatLngs. Also it allows to export/backup and iport/restore UNIQUES data to/from a local file.

### CODE REVIEW

File: plugins/uniques.js [hunk 1]
```diff
@@ -33,878 +33,878 @@ window.plugin.uniques.contentHTML = null;
...
+	var nick = window.PLAYER.nickname;
+	let match = false;
+	data.result.forEach(function(msg) {
+		match = false;
+		if (!window.plugin.uniques.parsedMsgs[msg[0]]){
+			var plext = msg[2].plext,
+				markup = plext.markup;
+
+		if(plext.plextType == 'SYSTEM_BROADCAST'
+		&& markup.length==5
```
Comment (le-jeu) [inline plugins/uniques.js]: the indentation here doesnt match

File: plugins/uniques.js [hunk 2]
```diff
@@ -552,7 +522,27 @@ window.plugin.uniques.highlighterDroned = {
 window.plugin.uniques.setupCSS = function() {
   $("<style>")
   .prop("type", "text/css")
-  .html('@include_string:uniques.css@')
```
Comment (le-jeu) [inline plugins/uniques.js]: either delete `uniques.css` or use it, using it sounds better

Comment (MysticJay) [reply plugins/uniques.js]: While I have no idea where the old code was introduced, the latest commit moves the code back to uniques.css

File: plugins/uniques.js [hunk 3]
```diff
@@ -510,45 +731,144 @@ window.plugin.uniques.onMissionLoaded = function(data) {
...
+		plugin.uniques.setPortalAction(wp.portal.guid,'visited');
 	});
 };
+/****************************************************************************************/
+/** Im-/Export of uniques ***************************************************************/
+/****************************************************************************************/
+window.plugin.uniques.optExport = function() {
+	var data = localStorage['plugin-uniques-data'];
+	window.saveFile(data, 'IITC-uniques.json', 'application/json');
+	confirm ("IITC-Uniques exported to IITC-uniques.json");
```
Comment (modos189) [inline plugins/uniques.js]: What if I press Cancel? And what changes if I press OK?
I propose to remove this line, similar to Draw tools - no such window appears there.

изображение

```suggestion
```

File: plugins/uniques.css [hunk 4]
```diff
@@ -17,3 +17,7 @@
   margin-top: -5px;
   margin-bottom: -5px;
 }
+
+.ui-dialog-portalslist {
+  max-width: none !important;
+}
```
Comment (johnd0e) [inline plugins/uniques.css]: This should not belong here

File: plugins/uniques.js [hunk 5]
```diff
@@ -1,9 +1,8 @@
 // @author         3ch01c
 // @name           Uniques
 // @category       Misc
-// @version        0.2.4
-// @description    Allow manual entry of portals visited/captured. Use the 'highlighter-uniques' plugin to show the uniques on the map, and 'sync' to share between multiple browsers or desktop/mobile. It will try and guess which portals you have captured from COMM/portal details, but this will not catch every case.
-
+// @version        0.4.0
```
Comment (johnd0e) [inline plugins/uniques.js:4]: Why skip 0.3?

File: plugins/uniques.js [hunk 6]
```diff
@@ -60,9 +63,12 @@ window.plugin.uniques.onPortalDetailsUpdated = function() {
 
 window.plugin.uniques.onPublicChatDataAvailable = function(data) {
 	var nick = window.PLAYER.nickname;
+	let match = false;
```
Comment (johnd0e) [inline plugins/uniques.js]: var

File: plugins/uniques.js [hunk 7]
```diff
@@ -167,9 +218,13 @@ window.plugin.uniques.updateCheckedAndHighlight = function(guid) {
 
 		var uniqueInfo = plugin.uniques.uniques[guid],
 			visited = (uniqueInfo && uniqueInfo.visited) || false,
-			captured = (uniqueInfo && uniqueInfo.captured) || false;
+			captured = (uniqueInfo && uniqueInfo.captured) || false,
+			scouted = (uniqueInfo && uniqueInfo.scouted) || false,
+			droned = (uniqueInfo && uniqueInfo.droned) || false;
```
Comment (johnd0e) [inline plugins/uniques.js]: Should be same names as in history.
Here and in numerous places below.

File: plugins/uniques.js [hunk 8]
```diff
@@ -179,42 +234,39 @@ window.plugin.uniques.updateCheckedAndHighlight = function(guid) {
...
+		window.plugin.uniques.missedLatLngs[id].action[action] = true;
+		if (action === 'captured') window.plugin.uniques.missedLatLngs[id].action.visited = true;
+		window.plugin.uniques.storeLocal('missedLatLngs');
 	}
-
-	plugin.uniques.updateCheckedAndHighlight(guid);
-	plugin.uniques.sync(guid);
 }
-
+// Visted
```
Comment (johnd0e) [inline plugins/uniques.js]: Typo. And why do we need this comment?

File: plugins/uniques.js [hunk 9]
```diff
@@ -238,7 +290,7 @@ window.plugin.uniques.updateVisited = function(visited, guid) {
 	plugin.uniques.updateCheckedAndHighlight(guid);
 	plugin.uniques.sync(guid);
 }
-
+// Captured
```
Comment (johnd0e) [inline plugins/uniques.js]: Needless?

File: plugins/uniques.js [hunk 10]
```diff
@@ -262,6 +314,50 @@ window.plugin.uniques.updateCaptured = function(captured, guid) {
 	plugin.uniques.updateCheckedAndHighlight(guid);
 	plugin.uniques.sync(guid);
 }
+// Scouted
```
Comment (johnd0e) [inline plugins/uniques.js]: Needless?

File: plugins/uniques.js [hunk 11]
```diff
@@ -262,6 +314,50 @@ window.plugin.uniques.updateCaptured = function(captured, guid) {
...
+	if (scouted) {
+		uniqueInfo.scouted = true;
+	} else {
+		uniqueInfo.scouted = false;
+	}
+
+	plugin.uniques.updateCheckedAndHighlight(guid);
+	plugin.uniques.sync(guid);
+}
+// Droned
```
Comment (johnd0e) [inline plugins/uniques.js]: Needless?

File: plugins/uniques.js [hunk 12]
```diff
@@ -262,6 +314,50 @@ window.plugin.uniques.updateCaptured = function(captured, guid) {
...
+	var uniqueInfo = plugin.uniques.uniques[guid];
+	if (!uniqueInfo) {
+		plugin.uniques.uniques[guid] = uniqueInfo = {
+			scouted: false,
+		};
+	}
+
+	if(scouted == uniqueInfo.scouted) return;
+
+	if (scouted) {
```
Comment (johnd0e) [inline plugins/uniques.js]: Space after `if`, here and everywhere.

File: plugins/uniques.js [hunk 13]
```diff
@@ -510,45 +731,143 @@ window.plugin.uniques.onMissionLoaded = function(data) {
...
+	window.saveFile(data, 'IITC-uniques.json', 'application/json');
+}
+window.plugin.uniques.optImport = function() {
+	L.FileListLoader.loadFiles({accept:'application/json'})
+		.on('load',function (e) {
+			try {
+				if (!confirm("Please only confirm this if you know what you are doing!!\nAre you sure you want to save your Unique visits/captures back to IITC?")) return;
+				var data = JSON.parse(e.reader.result);
+
+				if (Object.keys(data).length) window.plugin.uniques.uniques = data;
```
Comment (johnd0e) [inline plugins/uniques.js]:
```suggestion
				if (Object.keys(data).length) {
				  window.plugin.uniques.uniques = data;
				}
```

File: plugins/uniques.js [hunk 14]
```diff
@@ -510,45 +731,143 @@ window.plugin.uniques.onMissionLoaded = function(data) {
...
+/****************************************************************************************/
+window.plugin.uniques.optExport = function() {
+	var data = localStorage['plugin-uniques-data'];
+	window.saveFile(data, 'IITC-uniques.json', 'application/json');
+}
+window.plugin.uniques.optImport = function() {
+	L.FileListLoader.loadFiles({accept:'application/json'})
+		.on('load',function (e) {
+			try {
+				if (!confirm("Please only confirm this if you know what you are doing!!\nAre you sure you want to save your Unique visits/captures back to IITC?")) return;
```
Comment (johnd0e) [inline plugins/uniques.js]:
```suggestion
				var msgExit = 'Please only confirm this if you know what you are doing!\n'
				                    + 'Are you sure you want to save your Unique visits/captures back to IITC?';
				if (!confirm(msgExit)) {
				  return;
				}
```

(too long line)

File: plugins/uniques.js [hunk 15]
```diff
@@ -510,45 +731,143 @@ window.plugin.uniques.onMissionLoaded = function(data) {
...
+		});
+}
+
+/*****************************************************************************************/
+/** UNIQUES Backlog to be processed whenever a portal's LatLonE6 can be resolved to GUID */
+/*****************************************************************************************/
+window.plugin.uniques.onPortalAdded = function(data) {
+	let guid = data.portal.options.guid;
+	let p = data.portal.options.data;
+	let id = p.latE6 + "," + p.lngE6;
```
Comment (johnd0e) [inline plugins/uniques.js]: var

File: plugins/uniques.js [hunk 16]
```diff
@@ -510,45 +731,143 @@ window.plugin.uniques.onMissionLoaded = function(data) {
...
+		delete window.plugin.uniques.missedLatLngs[id];
+		window.plugin.uniques.storeLocal('missedLatLngs');
+		$("#missedPL").html(window.plugin.uniques.genList());
+	}
+}
 
+window.plugin.uniques.removeOldParsedMsgs = function() {
+//	remove all timestamps older than 30 days
+	let old = (Date.now() - (30*24*60*60*1000)); //30 days in miliseconds
+	let count = 0;
```
Comment (johnd0e) [inline plugins/uniques.js]: var

File: plugins/uniques.js [hunk 17]
```diff
@@ -557,15 +876,30 @@ var setup = function() {
...
-	
+
 	// to mark mission portals as visited
 	window.addHook('plugin-missions-mission-changed', window.plugin.uniques.onMissionChanged);
 	window.addHook('plugin-missions-loaded-mission', window.plugin.uniques.onMissionLoaded);
-	
+
+	// add controls to toolbox
+
+	link = $("<a onclick=\"window.plugin.uniques.options();return false\" title=\"Manage UNIQUES\">Uniques</a>");
```
Comment (johnd0e) [inline plugins/uniques.js]:
```suggestion
	var link = $('<a onclick="window.plugin.uniques.options();return false" title="Manage UNIQUES">Uniques</a>');
```

BUT!
Avoid js code in html attributes.
Replace it with jQuery listeners.
```suggestion
	var link = $('<a title="Manage UNIQUES">Uniques</a>')
	  .click(window.plugin.uniques.options);
```

**Review by johnd0e — CHANGES_REQUESTED** [review]
Very brief review for now.

File: plugins/uniques.js (hunk 5 above)
Comment (MysticJay) [reply plugins/uniques.js:4]: 0.3 was some intermediate release and I want to ensure that people get the update from release. so 0.4.0 was never used in either a draft or beta or RC and it will update.

Comment (johnd0e) [reply plugins/uniques.js:4]: > 0.3 was some intermediate release

As I can see we never had such release on official site.

>  I want to ensure that people get the update from release.

Do you mean auto-update routines?
They also rely on `@updateURL`, so it's not enough to change version.

File: plugins/uniques.js [hunk 18]
```diff
@@ -399,6 +496,48 @@ window.plugin.uniques.highlighter = {
...
+// highlighter scouted
+window.plugin.uniques.highlighterScouted = {
+	highlight: function(data) {
+		var guid = data.portal.options.ent[0];
+		var uniqueInfo = window.plugin.uniques.uniques[guid];
+
+		var style = {};
+
+		if (uniqueInfo && uniqueInfo.scouted)
+		{ // scouted - no highlights
```
Comment (johnd0e) [inline plugins/uniques.js]: https://eslint.org/docs/rules/brace-style (1tbs)

File: plugins/uniques.js [hunk 19]
```diff
@@ -399,6 +496,48 @@ window.plugin.uniques.highlighter = {
...
+// highlighter droned
+window.plugin.uniques.highlighterDroned = {
+	highlight: function(data) {
+		var guid = data.portal.options.ent[0];
+		var uniqueInfo = window.plugin.uniques.uniques[guid];
+
+		var style = {};
+
+		if (uniqueInfo && uniqueInfo.droned)
+		{ // droneded - no highlights
```
Comment (johnd0e) [inline plugins/uniques.js]: https://eslint.org/docs/rules/brace-style (1tbs)

File: plugins/uniques.js [hunk 20]
```diff
@@ -411,50 +550,71 @@ window.plugin.uniques.setupContent = function() {
...
+		var info = plugin.uniques.uniques[guid];
+		if (!info) return 0;
+		if (info.droned === undefined ) return 0;
+		if (info.droned === true) return 1;
 	}
 
-	window.plugin.portalslist.fields.push({
-		title: "Visit",
+	window.plugin.portalslist.fields.push(
+		{title: "V/C",
```
Comment (johnd0e) [inline plugins/uniques.js]: https://eslint.org/docs/rules/brace-style (1tbs)

File: plugins/uniques.js [hunk 21]
```diff
@@ -487,16 +647,77 @@ window.plugin.uniques.setupPortalsList = function() {
 					ev.preventDefault();
 					return false;
 				}, false);
+			},
 		},
-	});
-}
+//---------------------------------------------------------------------------
+		{title: "S",
```
Comment (johnd0e) [inline plugins/uniques.js]: `title` better move to next line, indented

File: plugins/uniques.js [hunk 22]
```diff
@@ -487,16 +647,77 @@ window.plugin.uniques.setupPortalsList = function() {
...
+				.appendTo(cell)
+				[0].addEventListener("change", function(ev) {
+					window.plugin.uniques.updateScouted(this.checked, guid);
+					ev.preventDefault();
+					return false;
+				}, false);
+		},
+	},
+//---------------------------------------------------------------------------
+		{title: "D",
```
Comment (johnd0e) [inline plugins/uniques.js]: Same here

File: plugins/uniques.css (hunk 4 above)
Comment (johnd0e) [reply plugins/uniques.css]: ?

**Review by johnd0e — CHANGES_REQUESTED** [review]

File: plugins/uniques.css (hunk 4 above)
Comment (MysticJay) [reply plugins/uniques.css]: We have to widen the portal-list dialog to avoid a horizontal scroll bar.
it was fixed to 600px for no obvious reason.
This is the only place we found to be appropriate

File: plugins/uniques.js (hunk 18 above)
Comment (MysticJay) [reply plugins/uniques.js]: done

File: plugins/uniques.js (hunk 19 above)
Comment (MysticJay) [reply plugins/uniques.js]: done

File: plugins/uniques.js (hunk 20 above)
Comment (MysticJay) [reply plugins/uniques.js]: reason: having the { in the same line as "title" allows to fold the section in the editor.
nevertheless moved the { back to previous line.

File: plugins/uniques.js (hunk 21 above)
Comment (MysticJay) [reply plugins/uniques.js]: done

File: plugins/uniques.js (hunk 22 above)
Comment (MysticJay) [reply plugins/uniques.js]: done

File: plugins/uniques.js (hunk 11 above)
Comment (MysticJay) [reply plugins/uniques.js]: personal flavor

File: plugins/uniques.js (hunk 10 above)
Comment (MysticJay) [reply plugins/uniques.js]: personal flavor

File: plugins/uniques.js (hunk 9 above)
Comment (MysticJay) [reply plugins/uniques.js]: personal flavor

**Review by MysticJay — COMMENTED** [review]
reviewed the change requests.
some accepted, some delayed to next release.

**Review by MysticJay — COMMENTED** [review]
code style fixes

File: plugins/uniques.css (hunk 4 above)
Comment (le-jeu) [reply plugins/uniques.css]: https://github.com/IITC-CE/ingress-intel-total-conversion/blob/67fcdf9adb8c079de6332742f12df7c99f4b91bb/core/style.css#L1036-L1044

Don't use `!important`
prefer use more specific rules like 
```css
.ui-dialog.ui-dialog-portallist {
  max-width: max-width: calc(100vw - 2px);
}
```

or open a discussion about the 600px

File: plugins/uniques.js [hunk 23]
```diff
@@ -139,10 +167,10 @@ window.plugin.uniques.onPublicChatDataAvailable = function(data) {
 		&& markup[3][0] == 'PORTAL'
 		&& markup[4][0] == 'TEXT'
 		&& markup[4][1].plain == ' has decayed') {
-		    // search for "Your Lx Resonator on y has decayed"
+				// search for "Your Lx Resonator on y has decayed"
```
Comment (johnd0e) [inline plugins/uniques.js]: Wrong indentation?

File: plugins/uniques.js [hunk 24]
```diff
@@ -154,8 +182,31 @@ window.plugin.uniques.onPublicChatDataAvailable = function(data) {
...
+				var portal = markup[1][1];
+				match = true;
+				plugin.uniques.setPortalAction(portal,'scouted');
+		} else if (plext.plextType == 'SYSTEM_NARROWCAST'
+			&& markup.length==3
+			&& markup[0][0] == 'TEXT'
+			&& markup[0][1].plain == 'You were displaced as Scout Controller on '
+			&& markup[1][0] == 'PORTAL') {
+				// search for "You were displaced as Scout Controller on"
+				var portal = markup[1][1];
```
Comment (johnd0e) [inline plugins/uniques.js]: Indentation of `&&` does not much similar cases in upper code

File: plugins/uniques.js [hunk 25]
```diff
@@ -154,8 +182,31 @@ window.plugin.uniques.onPublicChatDataAvailable = function(data) {
...
+				match = true;
+				plugin.uniques.setPortalAction(portal,'scouted');
+		}
+		}
```
Comment (johnd0e) [inline plugins/uniques.js:206]: Hm..
Hard to follow matching opening braces.

File: plugins/uniques.js (hunk 7 above)
Comment (johnd0e) [reply plugins/uniques.js]: Still unresolved. Properties, ids, ...

File: plugins/uniques.js [hunk 26]
```diff
@@ -179,84 +234,125 @@ window.plugin.uniques.updateCheckedAndHighlight = function(guid) {
...
+		if (action === 'captured') uniqueInfo.visited = true;
+		window.plugin.uniques.uniques[guid] = uniqueInfo;
+		window.plugin.uniques.storeLocal('uniques');
+		// trigger highlighters
+		plugin.uniques.updateCheckedAndHighlight(guid);
+		// triger sync
+		plugin.uniques.sync(guid);
+	} else { //guid not found, so add to missedLatLngs
+		if (!window.plugin.uniques.missedLatLngs[id])
+			window.plugin.uniques.missedLatLngs[id] = {portal:portal,action:{}};
```
Comment (johnd0e) [inline plugins/uniques.js:265]: Braces?

File: plugins/uniques.js [hunk 27]
```diff
@@ -179,84 +234,125 @@ window.plugin.uniques.updateCheckedAndHighlight = function(guid) {
...
+		if (!uniqueInfo) uniqueInfo = {};
+				// merge ALL pending actions, then remove from missedLatLngs
+		if (window.plugin.uniques.missedLatLngs[id]) {
+			Object.assign(uniqueInfo,window.uniques.missedLatLngs[id].action);
+			delete window.plugin.uniques.missedLatLngs[id];
+			window.plugin.uniques.storeLocal('missedLatLngs');
+		}
+		uniqueInfo[action] = true;
+		// special handling for captured
+		if (action === 'captured') uniqueInfo.visited = true;
```
Comment (johnd0e) [inline plugins/uniques.js]: I'd prefer to see braces even in single-line conditions. Here and in numerous other places.

File: plugins/uniques.js [hunk 28]
```diff
@@ -411,50 +550,71 @@ window.plugin.uniques.setupContent = function() {
 	plugin.uniques.contentHTML = '<div id="uniques-container">'
 		+ '<label><input type="checkbox" id="visited" onclick="window.plugin.uniques.updateVisited($(this).prop(\'checked\'))"> Visited</label>'
 		+ '<label><input type="checkbox" id="captured" onclick="window.plugin.uniques.updateCaptured($(this).prop(\'checked\'))"> Captured</label>'
+		+ '<label><input type="checkbox" id="scouted" onclick="window.plugin.uniques.updateScouted($(this).prop(\'checked\'))"> Scout controlled/label>'
+		+ '<label><input type="checkbox" id="droned" onclick="window.plugin.uniques.updateDroned($(this).prop(\'checked\'))"> Drone visited</label>'
```
Comment (johnd0e) [inline plugins/uniques.js]: Is it possible to replace inline js with jQuery listeners?

File: plugins/uniques.js [hunk 29]
```diff
@@ -510,45 +733,149 @@ window.plugin.uniques.onMissionLoaded = function(data) {
...
+/*****************************************************************************************/
+window.plugin.uniques.onPortalAdded = function(data) {
+	var guid = data.portal.options.guid;
+	var p = data.portal.options.data;
+	var id = p.latE6 + "," + p.lngE6;
+
+	//console.log("portal added: %s %s %o", guid, id, data);
+	// check if portal is in missedLatLngs
+	if (window.plugin.uniques.missedLatLngs[id]) {
+		console.log("found portal guid for previously missed portal: %s -> %s (%o)", id, guid, data);
```
Comment (johnd0e) [inline plugins/uniques.js]: No excessive logging

File: plugins/uniques.js [hunk 30]
```diff
@@ -510,45 +733,149 @@ window.plugin.uniques.onMissionLoaded = function(data) {
...
+	}
+	console.log ('[uniques] removed %s old parsedMsgs', count);
+}
+
+window.plugin.uniques.genList = function (){
+	var mLL = window.plugin.uniques.missedLatLngs;
+	var list = '';
+	for (var item in mLL) {
+		var p = mLL[item].portal;
+		list = list + '<a onclick=\"map.setView(['+p.latE6/1E6+','+p.lngE6/1E6+'],15);\">'+p.name+'</a><br>';
```
Comment (johnd0e) [inline plugins/uniques.js]:
```suggestion
		list = list + '<a onclick="map.setView(['+p.latE6/1E6+','+p.lngE6/1E6+'],15);">'+p.name+'</a><br>';
```
And is it possible to replace inline js with jQuery listeners?

File: plugins/uniques.js [hunk 31]
```diff
@@ -510,45 +733,149 @@ window.plugin.uniques.onMissionLoaded = function(data) {
...
+		if (aPortal.droned) droned++;
+	});
 
+	let list = 'Unique Portals Count:<br>visited: '+visited+' - captured: '+captured+'<br>scouted: '+scouted+' - droned: '+droned+'<br><hr>'
+			+ 'Missed Portals: <br>Click on portals to move the map to resolve the backlogged actions for this portal.<br><br>'
+			+ '<div id="missedPL" style="height:150px;overflow:auto">'
+			+ window.plugin.uniques.genList()
+			+ '</div><hr>'
+			+ '<a onclick=\"window.plugin.uniques.optExport();return false\" title=\"Export portals\' unique info to IITC.\">Backup</a> / '
+			+ '<a onclick=\"window.plugin.uniques.optImport();return false\" title=\"Import portals\' unique info to IITC.\">Restore</a> Uniques'
```
Comment (johnd0e) [inline plugins/uniques.js]:
```suggestion
			+ '<a onclick="window.plugin.uniques.optExport();return false" title="Export portals\' unique info to IITC.">Backup</a> / '
			+ '<a onclick="window.plugin.uniques.optImport();return false" title="Import portals\' unique info to IITC.">Restore</a> Uniques'
```
And is it possible to replace inline js with jQuery listeners?

**Review by johnd0e — CHANGES_REQUESTED** [review]
Original code seems overcomplicated...

File: plugins/uniques.js (hunk 7 above)
Comment (MysticJay) [reply plugins/uniques.js]: code and properties changed:
- scouted -> scoutControlled
- droned ->droneVisited

converter implemented. It will add a version info to the uniques MAP.

File: plugins/uniques.js (hunk 24 above)
Comment (MysticJay) [reply plugins/uniques.js]: now matching the same broken indentation as the upper code

File: plugins/uniques.js (hunk 30 above)
Comment (MysticJay) [reply plugins/uniques.js]: changed

File: plugins/uniques.js (hunk 29 above)
Comment (MysticJay) [reply plugins/uniques.js]: removed

File: plugins/uniques.js (hunk 31 above)
Comment (MysticJay) [reply plugins/uniques.js]: changed

File: plugins/uniques.css (hunk 4 above)
Comment (MysticJay) [reply plugins/uniques.css]: trying that

File: plugins/uniques.js [hunk 32]
```diff
@@ -411,50 +588,71 @@ window.plugin.uniques.setupContent = function() {
...
+	function droneVisitedValue(guid) {
+		var info = plugin.uniques.uniques[guid];
+		if (!info) return 0;
+		if (info.droneVisited === undefined ) return 0;
+		if (info.droneVisited === true) return 1;
 	}
 
 	window.plugin.portalslist.fields.push({
-		title: "Visit",
+		title: "V/C",
```
Comment (le-jeu) [inline plugins/uniques.js:648]: Visit/capture/scout will already be in portal-list (but drone/scanned aren't)
Don't you want to replace/edit the columns instead of duplicate them ?

File: plugins/uniques.js (hunk 28 above)
Comment (MysticJay) [reply plugins/uniques.js]: @johnd0e: on TG "So if this is not clear to you - better do nothing"

File: plugins/uniques.css [hunk 33]
```diff
@@ -18,6 +18,6 @@
   margin-bottom: -5px;
 }
 
-.ui-dialog-portalslist {
-  max-width: none !important;
-}
+.ui-dialog.ui-dialog-portallist {
+  max-width: max-width: calc(100vw - 2px);
+}
```
Comment (johnd0e) [inline plugins/uniques.css]: EOL expected

File: plugins/uniques.js (hunk 32 above)
Comment (johnd0e) [reply plugins/uniques.js:648]: > Don't you want to replace/edit the columns instead of duplicate them ?

Reuse, not replace.
E.g. use cached uniques data when there are no info from intel.
And highlight text in cases when data differ (with more explanation in tooltip).

File: plugins/uniques.js [hunk 34]
```diff
@@ -411,50 +588,71 @@ window.plugin.uniques.setupContent = function() {
 	plugin.uniques.contentHTML = '<div id="uniques-container">'
 		+ '<label><input type="checkbox" id="visited" onclick="window.plugin.uniques.updateVisited($(this).prop(\'checked\'))"> Visited</label>'
 		+ '<label><input type="checkbox" id="captured" onclick="window.plugin.uniques.updateCaptured($(this).prop(\'checked\'))"> Captured</label>'
+		+ '<label><input type="checkbox" id="scoutControlled" onclick="window.plugin.uniques.updateScoutControlled($(this).prop(\'checked\'))"> Scout controlled/label>'
```
Comment (le-jeu) [inline plugins/uniques.js]:
```suggestion
		+ '<label><input type="checkbox" id="scoutControlled" onclick="window.plugin.uniques.updateScoutControlled($(this).prop(\'checked\'))"> Scout controlled</label>'
```

**Review by le-jeu — CHANGES_REQUESTED** [review]
`IngressUniqueExporter.user.js` needs to be removed
and `README.md` restored

you can simply rebase and drop the related commits to keep only those that changes `uniques.{css,js}`

**Comment by Suburbanno** [comment]
@MysticJay Do you intend to continue this contribution? If not, could you close it?

**Comment by MysticJay** [comment]
please stop this
